### PR TITLE
Send page improvements to Zendesk

### PR DESCRIPTION
**Pull request by thehenster**
# Why

The service manual wants a way to allow users to give positive feedback, as well as negative.

# What

This receives feedback requests from the feedback frontend on behalf of the 'Improve this page' widget and forwards the feedback to Zendesk. https://trello.com/c/HN1L3EfK/286-update-footer-feedback-component

There isn't a requirement to browse the feedback anywhere apart from on Zendesk so we avoid the "Anonymous Feedback" flow in this app.

The 'Improve this page' feature is an experiment on the Service Manual. Traffic is expected to be very low. Therefore performance hasn't been taken seriously. Notably we are communicating with Zendesk synchronously. This will make the widget less snappy but has the advantage of being able to report to the user if it was successful.

### CODE REVIEW

File: spec/models/page_improvement_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      description: 'I love this page.',
+      name: 'John',
+      email: 'john@example.com',
+      user_agent: 'Safari',
+    })
+
+    expect(page_improvement.zendesk_ticket_attributes).to eq({
+      'subject' => '/service-manual/test',
+      'comment' => {
+        'body' => "[Details]\nI love this page.\n\n[Name]\nJohn\n\n[Email]\njohn@example.com\n\n[Path]\n/service-manual/test\n\n[User agent]\nSafari\n"
```
Comment (36degrees) [inline spec/models/page_improvement_spec.rb]: For the sake of readability and clarity on what we expect, is it worth making this and the body in the following block a heredoc?

File: spec/requests/page_improvements_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    expect(response.code).to eq('201')
+    expect(response_hash).to include('status' => 'success')
+  end
+
+  it "sends the feedback to Zendesk" do
+    zendesk_request = expect_zendesk_to_receive_ticket(
+      "subject" => "/service-manual/agile",
+      "comment" => {
+        "body" => "[Details]\nI have a problem\n\n[Name]\nJohn\n\n[Email]\njohn@example.com\n\n[Path]\n/service-manual/agile\n\n[User agent]\nSafari\n"
```
Comment (36degrees) [inline spec/requests/page_improvements_spec.rb]: Same thing re heredoc for clarity?

Comment (thehenster) [inline spec/requests/page_improvements_spec.rb]: Thanks. I've replaced both of these with heredocs.

**Comment by NeilvB** [comment]
Looks good 👍

**Pull request merged**